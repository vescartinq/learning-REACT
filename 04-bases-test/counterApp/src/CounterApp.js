import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 10 }) => {
  const [counter, setCounter] = useState(value); // []

  // handleAdd
  const handleAdd = () => {
    setCounter(counter + 1);
    // setCounter( (c) => c + 1 );
  };

  const handleSubtract = () => setCounter(counter - 1);

  const handleReset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2> {counter} </h2>

      <button id="plus" onClick={handleAdd}>
        +1
      </button>
      <button id="reset" onClick={handleReset}>
        Reset
      </button>
      <button id="less" onClick={handleSubtract}>
        -1
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};

export default CounterApp;
