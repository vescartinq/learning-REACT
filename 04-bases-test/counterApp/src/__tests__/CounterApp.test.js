import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import CounterApp from './../CounterApp';

describe('CounterApp', () => {
  let wrapper; //declaramos un contenedor en el scope general, pero que pueda ser variable
  // inicializado como undefined
  beforeEach(() => {
    wrapper = shallow(<CounterApp />); //inicializa el componente antes de cada test
  });

  test('should show CounterApp component working', () => {
    // const wrapper = shallow(<CounterApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show 100 by default', () => {
    const wrapper = shallow(<CounterApp value={100} />);

    const counterText = wrapper.find('h2').text().trim();

    expect(counterText).toBe('100');
  });

  test('should increment by 1 when click the plus button', () => {
    // const wrapper = shallow(<CounterApp />);

    // const btn1 = wrapper.find('button').at(0);
    // console.log(btn1.html()); //<button>+1</button>

    // ----EJECUTA EL CLICK-----
    // const btn1 = wrapper.find('#plus').at(0).simulate('click'); //at(0)-> Primer BotÃ³n
    const btn1 = wrapper.find('#plus').simulate('click');
    console.log(btn1.html());

    // ----LEE EL RESULTADO-----
    const counterText = wrapper.find('h2').text().trim();

    // ---CONFIRMA EL RESULTADO
    expect(counterText).toBe('11'); //10(valor inicial)+1
  });

  test('should decrease 1 when click the less button', () => {
    // const wrapper = shallow(<CounterApp />);

    const btn3 = wrapper.find('#less').simulate('click');
    console.log(btn3.html());

    const counterText = wrapper.find('h2').text().trim();

    expect(counterText).toBe('9'); //10(valor inicial)-1
  });

  test('should return to default value when click the reset button', () => {
    // const wrapper = shallow(<CounterApp />);

    wrapper.find('#plus').simulate('click'); //11
    wrapper.find('#plus').simulate('click'); //12

    const btn2 = wrapper.find('#reset').simulate('click'); //10
    console.log(btn2.html());

    const counterText = wrapper.find('h2').text().trim();

    expect(counterText).toBe('10'); //valor default
  });
});
