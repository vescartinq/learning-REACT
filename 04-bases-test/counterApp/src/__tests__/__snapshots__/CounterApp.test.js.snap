// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp should show CounterApp component working 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    id="plus"
    onClick={[Function]}
  >
    +1
  </button>
  <button
    id="reset"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    id="less"
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
